### PR TITLE
[O11y][Azure] Add global filter for dashboards to improve performance related to infraobs packages

### PR DESCRIPTION
**Pull request by milan-elastic**
- Enhancement

## Proposed commit message

Added global filter for below dashboards to improve performance,

- [Logs Azure] Alerts Overview  
- [Logs Azure] Cloud Overview 
- [Logs Azure] User Activity 
- [Logs Azure] Azure Spring Apps Logs System Logs
- [Logs Azure] Azure Spring Apps Logs Overview
- [Logs Azure] Azure Spring Apps Logs Application Console Logs

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates #10200 

## Screenshots

<details>
  <summary>[Logs Azure] Alerts Overview</summary>

| Before| After|
|:-:|:-:|
| before-azure-alerts-overview | after_azure_alert_overview |
</details>

<details>
  <summary>[Logs Azure] Cloud Overview</summary>

| Before| After|
|:-:|:-:|
| before_azure_acloud_overview | after_azure_cloud_overview |
</details>

<details>
  <summary>[Logs Azure] User Activity</summary>

| Before| After|
|:-:|:-:|
| before-azure-user activity log | after-azure-user activity log |
</details>

<details>
  <summary>[Logs Azure] Azure Spring Apps Logs System Logs</summary>

| Before| After|
|:-:|:-:|
| before-azure-spring-apps-logs-system-logs | after-azure-spring-apps-logs-system-logs |
</details>

<details>
  <summary>[Logs Azure] Azure Spring Logs Overview</summary>

| Before| After|
|:-:|:-:|
| before-azure-spring-apps-overview-log | after-azure-spring-apps-overview-log |
</details>

<details>
  <summary>[Logs Azure] Azure Spring Apps Logs Application Console Logs</summary>

| Before| After|
|:-:|:-:|
| before-spring-apps-logs-application-console-logs | after-spring-apps-logs-application-console-logs |
</details>

### CODE REVIEW

**Review by kush-elastic — APPROVED** [review]
LGTM!

File: packages/azure/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,8 @@
-
+- version: "1.14.0"
+  changes:
+    - description: Add global filter for dashboards to improve performance related to infraobs packages.
```
Comment (muthu-mps) [inline packages/azure/changelog.yml:3]:
```suggestion
    - description: Add global dataset filter for dashboards to improve performance.
```

File: packages/azure/kibana/dashboard/azure-0f559cc0-f0d5-11e9-90ec-112a988266d5.json [hunk 2]
```diff
@@ -485,21 +505,23 @@
...
     },
-    "coreMigrationVersion": "8.6.2",
-    "created_at": "2023-06-08T07:33:56.370Z",
+    "coreMigrationVersion": "8.8.0",
+    "created_at": "2024-04-19T09:37:54.808Z",
     "id": "azure-0f559cc0-f0d5-11e9-90ec-112a988266d5",
-    "migrationVersion": {
-        "dashboard": "8.6.0"
-    },
+    "managed": false,
```
Comment (muthu-mps) [inline packages/azure/kibana/dashboard/azure-0f559cc0-f0d5-11e9-90ec-112a988266d5.json:518]: I am hopping that the managed flag here is no way related to the managed tags.

**Review by muthu-mps — APPROVED** [review]
LGTM!

**Comment by harnish-elastic** [comment]
Pinging @elastic/security-service-integrations and @elastic/obs-ds-hosted-services team to provide CODEOWNER approval!

**Pull request merged**